File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int windowSum = 0;
        int count = 0;
        for(int i = 0; i<k;i++){
            windowSum+=arr[i];
        }
        if(windowSum/k >=threshold){
            count+=1;
        }
        int max = windowSum;
        for(int i = k; i < arr.length; i++){
            windowSum += arr[i];
            windowSum -= arr[i-k];
            if(windowSum/k >= threshold){
                count+=1;
            }
        }
        return count;
    }
}

File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        window_sum  = sum(arr[:k])
        count = 0
        if window_sum /k>=threshold:
            count+=1
        
        for i in range(k,len(arr)):
            window_sum  += arr[i] - arr[i-k]
            if window_sum /k >= threshold:
                count+=1
        return count